docs-src/api-overview.md

```diff
@@ -79,6 +96,128 @@ BackendPolicy. For more information on what may be configured with this resource
+Routes bind to Gateways is controlled by the resources themselves. Route and
+Gateway resources have built-in controls to permit or constrain how they select
+each other. This is useful for enforcing organizational policies for how Routes
+are exposed and on which Gateways. Consider the following example:
+
+> A Kubernetes cluster admin has deployed a Gateway “shared-gw” in the “Infra”
+> Namespace to be used by different application teams for exposing their
+> applications outside the cluster. Teams A and B (in Namespaces “A” and “B”
+> respectively) bind their Routes to this Gateway. They are unaware of each other
+> and as long as their Route rules do not conflict with each other they can
+> continue operating in isolation. Team C has special networking needs (perhaps
+> performance, security, or criticality) and they need a dedicated Gateway to
+> proxy their application to the outside world. Team C deploys their own Gateway 
+> “dedicated-gw”  in the “C” Namespace that can only be used by apps in the "C" 
+> Namespace.
+
+![schema](images/gateway-route-binding.png)
+
+There is a lot of flexibility in how Routes can bind to Gateways to achieve
+different organizational policies and scopes of responsibility. These are 
+different relationships that Gateways and Routes can have:
+
+- **One-to-one** - A Gateway and Route may be deployed and used by a single 
+  owner and have a one-to-one relationship. Team C is an example of this. This 
+  is also how the Ingress resource works.
+- **One-to-many** - A Gateway can have many Routes bound to it that are owned by
+  different teams from across different Namespaces. Teams A and B are an example 
+  of this.
+- **Many-to-one** - Routes can also be bound to more than one Gateway, allowing 
+  a single Route to control application exposure simultaneously across different
+  IPs, load balancers, or networks.
+
+*In summary, Gateways select Routes and Routes also select Gateways. When both 
+select each other, and there are no conflicts between existing bound Routes,
+then a Route will bind to a Gateway. When Routes are bound to a Gateway, it
+means their collective routing rules are configured on the underlying load
+balancers or proxies that are managed by that Gateway. Thus, a Gateway is a
+symbolic representation of a networking data plane that can be configured
+through Routes.*
+
+#### Route Selection
+
+The primary way in which Gateways select Routes (effectively allowing them to
+bind) is via their Route resource labels. Routes are actually bound to specific
+listeners within the Gateway so each listener has a `listener.routes` field
+which selects Routes by different criteria. There are three criteria considered
+for a Gateway to select Routes:
+
+- **Route label** - A Gateway selects Routes via labels that exist on the 
+resource (similar to how Services select Pods via Pod labels).
+- **Route type** - A Gateway listener can only select a single type of Route 
```

Comment:
nit: How about "Route Kind"?
We have chosen to use `kind` in all of our documentation and API as that's familiar to most users.

Reply:
makes sense - I've changed these headings to Label, Kind, and Namespace